google/cloud/internal/oauth2_service_account_credentials.cc

```diff
@@ -65,13 +67,15 @@ StatusOr<ServiceAccountCredentialsInfo> ParseServiceAccountCredentials(
                         " field is empty on data loaded from " + source);
     }
   }
-  // The token_uri field may be missing, but may not be empty:
-  if (credentials.count(token_uri_key) != 0 &&
-      credentials.value(token_uri_key, "").empty()) {
-    return Status(StatusCode::kInvalidArgument,
-                  "Invalid ServiceAccountCredentials, the " +
-                      std::string(token_uri_key) +
-                      " field is empty on data loaded from " + source);
+  // The token_uri and universe_domain fields may be missing, but may not be
+  // empty.
+  for (auto const& key : {token_uri_key, universe_domain_key}) {
+    if (credentials.count(key) != 0 && credentials.value(key, "").empty()) {
+      return Status(StatusCode::kInvalidArgument,
```

Comment:
Consider using `internal::InvalidArgumentError(...)` here, since you now include `make_status.h`.